src/NHibernate/AdoNet/HanaBatchingBatcher.cs

```diff
@@ -0,0 +1,173 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Text;
+using NHibernate.AdoNet.Util;
+using NHibernate.Exceptions;
+
+namespace NHibernate.AdoNet
+{
+	/// <summary>
+	/// Summary description for HanaBatchingBatcher.
+	/// By Jonathan Bregler
+	/// </summary>
+	public partial class HanaBatchingBatcher : AbstractBatcher
+	{
+		private int _batchSize;
+		private int _countOfCommands;
+		private int _totalExpectedRowsAffected;
+		private DbCommand _currentBatch;
+		private readonly IList<DbCommand> _currentBatchCommands = new List<DbCommand>();
+		private StringBuilder _currentBatchCommandsLog;
+
+		public HanaBatchingBatcher(ConnectionManager connectionManager, IInterceptor interceptor)
+			: base(connectionManager, interceptor)
+		{
+			_batchSize = Factory.Settings.AdoBatchSize;
+			//we always create this, because we need to deal with a scenario in which
+			//the user change the logging configuration at runtime. Trying to put this
+			//behind an if(log.IsDebugEnabled) will cause a null reference exception 
+			//at that point.
+			_currentBatchCommandsLog = new StringBuilder().AppendLine("Batch commands:");
+		}
+
+		public override void AddToBatch(IExpectation expectation)
+		{
+			Debug.Assert(CurrentCommand is ICloneable); // HanaCommands are cloneable
```

Comment:
Well, there are some other cases, but there are likewise many things we are changing/modernizing/fixing, and so we avoid new occurrences to be added. Another case is the Java naming, we try to avoid having more of it to be added and ask for things to be renamed according to .Net usual conventions. (It appears private field are no more covered by public guidelines, but previously they were.)